lib/Service/SignatureService.php

```diff
@@ -225,9 +269,15 @@ private function checkRequestSignature(IRequest $request): string {
 		$signed = base64_decode($sign['signature']);
 		$estimated = $this->generateEstimatedSignature($headers, $request);
 
+		$algorithm = 'sha256';
+		if ($this->get('algorithm', $sign, '') === 'rsa-sha256') {
```

Comment:
Possibly you could check for ```'rsa-sha512'``` here as well, like we do it here:
https://github.com/friendica/friendica/blob/2018.12-rc/src/Util/HTTPSignature.php#L421-L427

Reply:
should work now